src/foam/nanos/controller/ApplicationController.js

```diff
@@ -496,13 +499,15 @@ foam.CLASS({
       return self.crunchController.maybeLaunchInterceptView(intercept);
     },
 
-    function notify(data, type, description) {
-      /** Convenience method to create toast notifications. */
-      this.add(this.NotificationMessage.create({
-        message: data,
-        type: type,
-        description: description
-      }));
+    function notify(toastMessage, toastSubMessage, severity, transient) {
+      var notification = this.Notification.create();
+      notification.userId = this.user.id;
```

Comment:
Who is user? does this need to be realUser?

Reply:
Needs to be the logged in user, since this notify method is for showing in app notification messages. When a Notification is created from an approval request it creates a notification and sets the id to the approver which would send them a notification message once I get websockets working for the notificationDAO.

Reply:
yes, so where is it coming from? I kept expanding, but wasn't sure. Are you using subject.realUser or is it being set elsewhere?

Reply:
Does this work?

```suggestion
      notification.userId = this.subject.realUser.id;
```

Reply:
or 
```suggestion
      notification.userId = this.subject && this.subject.realUser ? this.subject.realUser.id : this.user.id;
```